src/controllers/restful/PaymentController.ts

```diff
@@ -410,25 +410,46 @@ export function getResponseErrorIfExists(
   if (faultBean === undefined) {
     return undefined;
   }
-  // Response is FAILED and additional information are provided by PagoPa
   return getErrorMessageCtrlFromPagoPaError(
+    faultBean.id,
     faultBean.faultCode,
-    faultBean.description
+    faultBean.description,
+    faultBean.originalFaultCode
   );
 }
 
 /**
  * Convert PagoPa message error (faultCode) to Controller message error (ErrorMessagesCtrlEnum) to send to BackendApp
  * A complete list of faultCode provided by PagoPa is available at
  * https://www.agid.gov.it/sites/default/files/repository_files/specifiche_attuative_nodo_2_1_0.pdf
+ * @param {string} faultId - Error Id provided by PagoPa to
  * @param {string} faultCode - Error code provided by PagoPa
  * @return {PaymentFaultEnum} Error code to send to BackendApp
  */
 export function getErrorMessageCtrlFromPagoPaError(
+  faultId: string,
   faultCode: string,
-  faultDescription: string | undefined
+  faultDescription: string | undefined,
+  originalFaultCode: string | undefined
 ): PaymentFaultEnum {
   switch (faultCode) {
+    case "PPT_EMESSO_DA_PAA":
+      return getErrorMessageFromPA(originalFaultCode);
```

Comment:
Different from previous implementation **getErrorMessageCtrlFromPagoPaError** checks on `faultCode` and **getErrorMessageFromPA** checks on `originalFaultCode`
What is exactly the difference of these 2 values?

Reply:
`faultCode` refers to error from the point of view of pagoPA. 
`orginalFaultCode` refers to error from the point of view of PA

pagoPA can raise error based on message format, but it doesn't know anything about the payment status. To have a detailed error, the `orginalFaultCode` is needed